### PR TITLE
Reorder sections to make it more clear to the reader that requesting a CVE number is optional

### LINKED ISSUE #10209: Make GHSA and CVE publication more distinct
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/code-security/security-advisories/publishing-a-security-advisory

### What part(s) of the article would you like to see updated?

I would like that the CVE number be made more clearly optional.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by darakian**
At the moment the publication doc leads with the topic of requesting a CVE number.
This PR moves things around and adds language to imply that a CVE number is not required in order to publish a GHSA.
The impetus for this PR comes from an out of band conversation with the rustsec community and some confusion around GHSA publication.

### Why:

Closes https://github.com/github/docs/issues/10209

### What's being changed:
Minor reordering of sections and more additional language.

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

File: content/code-security/security-advisories/publishing-a-security-advisory.md
Comment (rschultheis) [inline content/code-security/security-advisories/publishing-a-security-advisory.md:81]: If the goal is to emphasize the CVE being optional, then I think it might also be worth mentioning that our CNA should only be assigning to software not covered by other CNAs.


```suggestion
Github can provide a CVE number if needed and the software is not under the scope of another CVE Numbering Authority (CNA). Anyone with admin permissions to a security advisory can request a CVE identification number for the security advisory. If the advisory already has a CVE number, then that can be input in the advisory form.
```

**Review by rschultheis — DISMISSED** [review]
This is awesome, I left one suggestion though it might be a bit wordy.  I think this PR could go out as is.  Thanks @darakian

**Comment by ramyaparimi** [comment]
@darakian 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/code-security/security-advisories/publishing-a-security-advisory.md
Comment (lucascosti) [inline content/code-security/security-advisories/publishing-a-security-advisory.md:81]: 👋 @darakian, thanks heaps for this PR! I agree that if the step is optional, then it should be moved lower and marked as such. 

🤔 Most of the added content in this paragraph is already provided in the next paragraph in the `{% data reusables.repositories.request-security-advisory-cve-id %}` reusable. You can see the content of that resuable file [here](https://github.com/github/docs/blob/d1049f143a59a616a834cde9ea27722d6fcda77c/data/reusables/repositories/request-security-advisory-cve-id.md). You can see a [built stage of this branch here](https://docs-10210--reorder-advisory-p.herokuapp.com/en/code-security/security-advisories/publishing-a-security-advisory#requesting-a-cve-identification-number-optional).

I would recommend sticking with the current wording in the reusable, or modifying the reusable directly.

```suggestion
Anyone with admin permissions to a security advisory can request a CVE identification number for the security advisory.
```

Comment (darakian) [reply content/code-security/security-advisories/publishing-a-security-advisory.md]: Thanks for the tip. I was unaware of what those `{%...%}` patterns were. I've pushed a commit which removes the leading text from the base doc and alters the reusable. Let me know what you think @lucascosti 

@rschultheis can I get a second review on the updated content?

**Review by rschultheis — DISMISSED** [review]
this all looks good to me, though I defer to @lucascosti and docs team in general.  I think this does make it more clear that a CVE is optional which is the goal.

File: data/reusables/repositories/request-security-advisory-cve-id.md
Comment (lucascosti) [inline data/reusables/repositories/request-security-advisory-cve-id.md:1]: A few rewordings, plus I think it's important to keep the mention that requesting a CVE doesn't make things public.

```suggestion
If you want a CVE identification number for the security vulnerability in your project, and don't already have one, you can request a CVE identification number from {% data variables.product.prodname_dotcom %}. {% data variables.product.prodname_dotcom %} usually reviews the request within 72 hours. Requesting a CVE identification number doesn't make your security advisory public. If your security advisory is eligible for a CVE, {% data variables.product.prodname_dotcom %} will reserve a CVE identification number for your advisory. We'll then publish the CVE details after you make your security advisory public. Anyone with admin permissions to a security advisory can request a CVE identification number.
```

File: data/reusables/repositories/request-security-advisory-cve-id.md
Comment (lucascosti) [inline data/reusables/repositories/request-security-advisory-cve-id.md:3]:
```suggestion
If you already have a CVE you want to use, for example, if you use a CVE Numbering Authority (CNA) other than {% data variables.product.prodname_dotcom %}, add the CVE to the security advisory form. This may happen, for example, if you want to get the advisory consistent with other communications you plan to send out at publication time. {% data variables.product.prodname_dotcom %} cannot assign CVEs to your project if it is covered by another CNA.
```

**Review by lucascosti — COMMENTED** [review]
Thanks, @darakian! I've added a few wording suggestions.

🤔 I'm also having second thoughts about moving the "Requesting a CVE identification number" section below the "Publishing" one. If someone wants to have a CVE, does a CVE need to be assigned before an advisory is published?

If so, we'd probably want the articles and sections to be in the order of the workflow. e.g. Draft -> request CVE -> publish.

I think adding the `(Optional)` to the section title now clearly indicates that it is not required to publish an advisory.

**Comment by darakian** [comment]
> does a CVE need to be assigned before an advisory is published?

No. A CVE can be requested for the advisory either during the initial advisory publication step or later on.

> I've added a few wording suggestions.

Those look good to me 👍

**Review by lucascosti — APPROVED** [review]
> No. A CVE can be requested for the advisory either during the initial advisory publication step or later on.

Awesome, thanks for confirming! I'll merge this in 🚀

**Comment by hieueuro** [comment]
https://github.com/0386861680/docs/blob/5a1073477fd50b239dab8980e401b7c89d4350c4/CODE_OF_CONDUCT.md

**Pull request merged**
